src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1707,4 +1709,41 @@ private function addNotifierSection(ArrayNodeDefinition $rootNode)
             ->end()
         ;
     }
+
+    private function addRateLimiterSection(ArrayNodeDefinition $rootNode)
+    {
+        $rootNode
+            ->children()
+                ->arrayNode('rate_limiter')
+                    ->info('Rate limiter configuration')
+                    ->{!class_exists(FullStack::class) && class_exists(TokenBucketLimiter::class) ? 'canBeDisabled' : 'canBeEnabled'}()
+                    ->fixXmlConfig('limiter')
+                    ->beforeNormalization()
+                        ->ifTrue(function ($v) { return \is_array($v) && !isset($v['limiters']) && !isset($v['limiter']); })
+                        ->then(function (array $v) { return ['limiters' => $v]; })
+                    ->end()
+                    ->children()
+                        ->arrayNode('limiters')
```

Comment:
```
framework:
    rate_limiter:
        limiters:
            # allow up to 5 failed login attempts, before allowing 1 every 15 minutes
            login:
```

The `rate_limiter` and THEN `limiters` looks unfortunate. Is it possible to flatten it?

```
framework:
    rate_limiters:
        # allow up to 5 failed login attempts, before allowing 1 every 15 minutes
        login:
```

I can't think of any "system-wide rate limiter" config that might be needed in the future... but tell me if you disagree.

Reply:
Oh, in fact this config does allow skipping the "limiters". So:

```yaml
framework:
    rate_limiter:
        login:
            # ...
```

It can be usefull for forward-compatibility and XML-support:

```xml
<framework:config>
    <framework:rate-limiter>
        <framework:limiter name="..." strategy="..." limit="5"/>
        <framework:limiter name="..." strategy="..." limit="5"/>
    </framework:rate-limiter>
</framework:config>
```

Reply:
having a sub-node for the prototyped array is useful only for forward-compat to allow having other config settings in `rate_limiter`. XML compat is not an issue. Having `rate_limiters` in YAML would translation to multiple `<framework:rate-limiter>` nodes

Reply:
Everything is fine like this then, right?

Reply:
Yes, I think so.

@stof yes, but imho it's nicer to have one "main" element (`<rate-limiter>`) with multiple sub elements :)